lib/ansible/module_utils/network/exos/config/vlans/vlans.py

```diff
@@ -0,0 +1,271 @@
+        """
+        result = {'changed': False}
+        warnings = list()
+        requests = list()
+
+        existing_vlans_facts = self.get_vlans_facts()
+        requests.extend(self.set_config(existing_vlans_facts))
+        if requests:
+            if not self._module.check_mode:
+                send_requests(self._module, requests=requests)
+            result['changed'] = True
+        result['requests'] = requests
+
+        changed_vlans_facts = self.get_vlans_facts()
+
+        result['before'] = existing_vlans_facts
+        if result['changed']:
+            result['after'] = changed_vlans_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_vlans_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        have = existing_vlans_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        state = self._module.params['state']
+        if state == 'overridden':
+            requests = self._state_overridden(want, have)
+        elif state == 'deleted':
+            requests = self._state_deleted(want, have)
+        elif state == 'merged':
+            requests = self._state_merged(want, have)
+        elif state == 'replaced':
+            requests = self._state_replaced(want, have)
+        return requests
+
+    def _state_replaced(self, want, have):
+        """ The command generator when state is replaced
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        requests = []
+
+        request_patch = deepcopy(self.VLAN_PATCH)
+        request_body = deepcopy(self.REQUEST_BODY)
+
+        for w in want:
+            if w.get('vlan_id'):
+                h = search_obj_in_list(w['vlan_id'], have, 'vlan_id')
+                if h:
+                    if dict_diff(w, h):
+                        request_body = deepcopy(self.REQUEST_BODY)
+                        request_body = self._update_vlan_config_body(w, request_body)
+                        request_patch["data"]["openconfig-vlan:vlans"]["vlan"].append(request_body)
+                else:
+                    request_post = deepcopy(self.VLAN_POST)
```

Comment:
You have identical code on lines 159-163, 193-197, 228-232... Move it into a function.